.github/workflows/build-docker-image.yml

```diff
@@ -1,40 +1,63 @@
 name: Build multi-arch Docker Image
 
 on:
+  release:
+    types: [ published ]
   push:
-    branches: [main]
+    branches: [ main ]
   pull_request:
 
 jobs:
-  build:
+  build-linux-and-windows-multiarch:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-go@v3
-      with:
-        go-version: '1.18'
-    - run: go version
-
-    - name: Verify project
-      run: make verify
-
-    - name: Set up QEMU
-      uses: docker/setup-qemu-action@v2
-    - name: Set up Docker Buildx
-      id: buildx
-      uses: docker/setup-buildx-action@v2
-
-    - name: Docker info
-      run: docker info
-    - name: Buildx inspect
-      run: docker buildx inspect
-
-    - name: Build image
-      uses: docker/build-push-action@v3
-      with:
-        context: .
-        file: ./Dockerfile
-        platforms: linux/amd64,linux/arm/v7,linux/arm64,linux/ppc64le,linux/s390x
-        push: false
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.18'
+      - run: go version
+
+      - name: modVerify
+        run: go mod verify
+
+      - name: modTidy
+        run: go mod tidy
+
+      - name: ensure compilation
+        env:
+          GOOS: linux
+        run: go build
+
+      - name: Test
+        run: go test -v ./...
+
+      - name: Login to Docker Hub
+        # Only if we need to push an image
+        if: ${{ github.event_name == 'release' && github.event.action == 'published' }}
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      # Setup for buildx
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker info
+        run: docker info
+      - name: Buildx inspect
+        run: docker buildx inspect
+
+      # In case linux/s390x fails with tls issues:
+      # see https://github.com/testcontainers/moby-ryuk/pull/26
+      # and https://github.com/testcontainers/moby-ryuk/pull/40
+      - name: Cross Build for Linux and Windows
         # Use a 'temp' tag, that won't be pushed, for non-release builds
-        tags: testcontainers/ryuk:${{ github.event.release.tag_name || 'temp' }}
+        run: |
+          ./build-multiarch.sh testcontainers/ryuk:${{ github.event.release.tag_name || 'temp' }} ${{ (github.event_name == 'release' && github.event.action == 'published' && 'yes') || 'no' }}
```

Comment:
```suggestion
        run: |
          ./build-multiarch.sh testcontainers/ryuk:${RYUK_VERSION} ${IS_RELEASE}
          env:
            RYUK_VERSION: ${{ github.event.release.tag_name || 'temp' }}
            IS_RELEASE: ${{ (github.event_name == 'release' && github.event.action == 'published' && 'yes') || 'no' }}
```

Reply:
Indeed, and thanks for participating!

Could you move the `env` block section above run, for readability?